### PR TITLE
feat: upgrade `GasPriceOracle` in MPT parent block

### PR DESCRIPTION
**Pull request by seolaoh**
# Description

Upgrade `GasPriceOracle` in MPT parent block for changing `L1Block` address it refers.
Note that we don't consider the case that Ecotone upgrade is occurred after MPT hardfork.

### CODE REVIEW

File: packages/contracts/tasks/check-l2.ts
Comment (sm-stack) [inline packages/contracts/tasks/check-l2.ts:1]: Don't we have to add checks for newly deployed contracts?

File: op-node/rollup/derive/kromampt_upgrade_transactions.go [hunk 1]
```diff
@@ -31,15 +36,21 @@ var (
...
+	updateGasPriceOracleMPTProxySource = UpgradeDepositSource{Intent: "KromaMPT: Gas Price Oracle Proxy Update"}
+	enableKromaMPTSource               = UpgradeDepositSource{Intent: "KromaMPT: Gas Price Oracle Set Kroma MPT"}
+
+	enableKromaMPTInput = crypto.Keccak256([]byte("setKromaMPT()"))[:4]
 
 	l1BlockMPTDeploymentBytecode        = common.FromHex(bindings.L1BlockMetaData.Bin)
 	sequencerFeeVaultDeploymentBytecode = common.FromHex(bindings.ProtocolVaultMetaData.Bin)
 	baseFeeVaultDeploymentBytecode      = common.FromHex(bindings.ProtocolVaultMetaData.Bin)
 	l1FeeVaultDeploymentBytecode        = common.FromHex(bindings.L1FeeVaultMetaData.Bin)
+	gasPriceOracleMPTDeploymentBytecode = common.FromHex(bindings.GasPriceOracleMetaData.Bin)
```
Comment (sm-stack) [inline op-node/rollup/derive/kromampt_upgrade_transactions.go:49]: Could you fix this to the actual bytecode? Relevant comment [here](https://github.com/kroma-network/kroma/pull/398#discussion_r1881229723).

File: packages/contracts/contracts/test/Encoding.t.sol [hunk 2]
```diff
@@ -87,8 +87,8 @@ contract Encoding_Test is CommonTest {
 
         // assert KromaDepositTx
         {
-            bytes memory txn = Encoding.encodeDepositTransaction(t, true);
-            bytes memory _txn = ffi.encodeDepositTransaction(t, true);
+            txn = Encoding.encodeDepositTransaction(t, true);
+            _txn = ffi.encodeDepositTransaction(t, true);
 
             assertEq(txn, _txn, "failed to assert kroma deposit tx");
         }
```
Comment (Pangssu) [inline packages/contracts/contracts/test/Encoding.t.sol:94]: Remove bracket

File: packages/contracts/.gas-snapshot [hunk 3]
```diff
@@ -116,9 +116,8 @@ ColosseumTest:test_proveFaultWithZkEvm_succeeds() (gas: 2629287)
 ColosseumTest:test_proveFaultWithZkVm_succeeds() (gas: 2631088)
 ColosseumTest:test_proveFault_cancelChallenge_succeeds() (gas: 3086364)
 ColosseumTest:test_proveFault_finalizedOutput_reverts() (gas: 764729)
-Colosseum_MptTransition_Test:test_createChallenge_mptFirstOutputIndex_reverts() (gas: 4938257)
-Colosseum_MptTransition_Test:test_createChallenge_succeeds() (gas: 457498)
-Colosseum_MptTransition_Test:test_createChallenge_upgradeMptFirstOutputIndex_succeeds() (gas: 5320981)
+Colosseum_MptTransition_Test:test_createChallenge_mptFirstOutputIndex_reverts() (gas: 74743)
```
[the remark this answers is not in the document]
Comment (seolaoh) [reply packages/contracts/.gas-snapshot:119]: It seems not be updated before PR I think. You can refer other createChallenge revert tests gas usage.

File: op-node/rollup/derive/kromampt_upgrade_transactions_test.go [hunk 4]
```diff
@@ -3,11 +3,12 @@ package derive
...
 
-	"github.com/stretchr/testify/require"
-
-	"github.com/ethereum-optimism/optimism/op-bindings/predeploys"
+	oppredeploys "github.com/ethereum-optimism/optimism/op-bindings/predeploys"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/stretchr/testify/require"
+
+	"github.com/kroma-network/kroma/kroma-bindings/predeploys"
```
[the remark this answers is not in the document]
Comment (seolaoh) [reply op-node/rollup/derive/kromampt_upgrade_transactions_test.go:11]: I think we allow both rules for import, grouping only kroma or grouping kroma & optimism.

File: packages/contracts/contracts/L2/ProtocolVault.sol [hunk 5]
```diff
@@ -6,7 +6,9 @@ import { ISemver } from "../universal/ISemver.sol";
 
 /**
  * @custom:proxied
- * @custom:predeploy 0x4200000000000000000000000000000000000006
+ * @custom:predeploy 0x4200000000000000000000000000000000000006 (before Kroma MPT transition)
+ * @custom:predeploy 0x4200000000000000000000000000000000000011 (for SequencerFeeVault after Kroma MPT transition)
+ * @custom:predeploy 0x4200000000000000000000000000000000000019 (for BaseFeeVault after Kroma MPT transition)
```
Comment (Pangssu) [inline packages/contracts/contracts/L2/ProtocolVault.sol:11]: How about copying this file to create SequencerFeeVault and BaseFeeVault?

**Review by Pangssu — APPROVED** [review]

File: packages/contracts/contracts/test/GasPriceOracle.t.sol
Comment (sm-stack) [inline packages/contracts/contracts/test/GasPriceOracle.t.sol:1]: How do you think about adding a test for `setKromaMPT()`?

**Review by sm-stack — APPROVED** [review]

**Review by kangsorang — APPROVED** [review]
LGTM

**Pull request merged**